tooling/api/src/main/java/org/jboss/windup/tooling/ExecutionBuilderSetOptionsAndProgressMonitor.java

```diff
@@ -0,0 +1,36 @@
+package org.jboss.windup.tooling;
+
+import org.jboss.windup.exec.WindupProgressMonitor;
+
+/**
+ * Allows setting windup options, including the {@link WindupProgressMonitor}.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public interface ExecutionBuilderSetOptionsAndProgressMonitor
+{
+    /**
+     * Sets a pattern of file paths to ignore during processing.
+     */
+    ExecutionBuilderSetOptions ignore(String ignorePattern);
+
+    /**
+     * Sets the package name prefixes to scan (the default is to scan all packages).
+     */
+    ExecutionBuilderSetOptions includePackage(String includePackagePrefix);
```

Comment:
Yeah, I think we will want a different interface for non-Java tooling in the future. I figured that we can do that when the issue arises, though. At the moment, this is really all about Java tooling.

I'd like for the consumer of this API not to have to know about things like the generic options framework at the moment.